libuuid/src/uuid_time.c

```diff
@@ -75,6 +76,47 @@ time_t uuid_time(const uuid_t uu, struct timeval *ret_tv)
 	return tv.tv_sec;
 }
 
+#if defined(__USE_TIME_BITS64) && defined(__GLIBC__)
+#undef uuid_time
+
+struct timeval32
+{
+	int32_t tv_sec;
+	int32_t tv_usec;
+};
+
+/* prototype to make compiler happy */
+int32_t uuid_time(const uuid_t uu, struct timeval32 *ret_tv);
+
+/* Check whether T fits in time_t.  */
+static inline int
+in_time_t_range (time_t t)
+{
+  int32_t s = t;
+  return s == t;
+}
+
+int32_t uuid_time(const uuid_t uu, struct timeval32 *ret_tv)
+{
+	struct timeval		tv;
+	time_t ret_time = uuid_time64(uu, &tv);
```

Comment:
This is not too readable. Where is uuid_time64()? (yes, see the #define, but ...)

Could it be the code in uuid_time.c explicit and easy to read? I mean keep in the code two functions  __uuid_time32() and  __uuid_time64(), and later export to API what is necessary based on __USE_TIME_BITS64? 

(Maybe use weak-alias attribute or so, I'm not sure, glibc uses __REDIRECT).

Reply:
Ok, I used weak aliases now. Seems to work with my test binaries.